### ISSUE TITLE
RCS1220 incorrectly suggested when type being tested against is different from type being cast to; fix results in uncompilable code

### ISSUE THREAD

**Issue by IanKemp** [JosefPihrt/Roslynator#895]
**Product and Version Used**: 
VS 2022 17.1.5
Roslynator 2022 4.1.0

**Steps to Reproduce**:
Sample code:

```csharp
public static bool DoesPropertyExist(dynamic @object, string name)
{
	if (@object is ExpandoObject) // RCS1220 triggers here
		return ((IDictionary<string, object>)@object).ContainsKey(name);

	return @object.GetType().GetProperty(name) != null;
}
```

**Actual Behavior**:
RCS1220 is triggered and applying the codefix results in:

```csharp
public static bool DoesPropertyExist(dynamic @object, string name)
{
	if (@object is ExpandoObject expandoObjects)
		return (expandoObjects).ContainsKey(name);

	return @object.GetType().GetProperty(name) != null;
}
```

which no longer compiles.

**Expected Behavior**:
RCS1220 is not triggered.
